biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -1931,10 +1929,13 @@ subroutine count_cohorts( currentPatch )
        currentCohort => currentCohort%shorter
     enddo
 
-    if (backcount /= currentPatch%countcohorts) then
-       write(fates_log(),*) 'problem with linked list, not symmetrical'
-    endif
-
+    if(debug) then
+       if (backcount /= currentPatch%countcohorts) then
+          write(fates_log(),*) 'problem with linked list, not symmetrical'
+          call endrun(msg=errMsg(sourcefile, __LINE__))
+       endif
+    end if
```

Comment:
I don't think we need this check all the time, but don't feel strongly.  Since debug is a parameter constant, I think the compilers realize this. And such the logic doesn't get evaluated, which is a big time save.